packages/playground/data-liberation/src/entity-readers/WP_WXR_Sorted_Entity_Reader.php

```diff
@@ -0,0 +1,669 @@
+<?php
+
+use WordPress\ByteReader\WP_Byte_Reader;
+
+/**
+ * Data Liberation API: WP_WXR_Sorted_Entity_Reader class
+ *
+ * The topological sorted WXR reader class. This is an extension of the
+ * WP_WXR_Entity_Reader class that emits entities sorted topologically so that
+ * the parents are always emitted before the children.
+ *
+ * ## Implementation
+ *
+ * We create a custom table that contains the IDs and the new IDs created in the
+ * target system sorted in the parent-child order.
+ *
+ * This class extends the WP_WXR_Entity_Reader class and overrides the
```

Comment:
Neat idea! I initially thought about ingraining the topological sorting knowledge in the streaming importer to apply it to all possible data sources, but thinking about that more, how many other data sources might yield data in a child-first order? In the future – probably some. Today – I can't think of many. I kind of like keeping and maturing this logic in the entity reader class. Even if we need to move it to the importer one day, the API will mostly stay the same. Cool!